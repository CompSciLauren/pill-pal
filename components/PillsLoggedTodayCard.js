import React from 'react';
import PropTypes from 'prop-types';
import { Platform, StyleSheet, Text, View } from 'react-native';
import { FontAwesome } from '@expo/vector-icons';

export function PillsLoggedTodayCard(props) {
  const { title, infoArray } = props;

  return (
    <View style={styles.pillsLoggedInfoContainer}>
      <View style={{ flexDirection: 'row' }}>
        <View
          style={{
            marginLeft: 30,
            flexDirection: 'column',
            alignItems: 'center',
            width: '78%',
          }}
        >
          <Text style={styles.pillsLoggedTitleText}>{title}</Text>
        </View>
        <View
          style={{
            flexDirection: 'column',
            alignItems: 'flex-start',
            width: '22%',
          }}
        >
          <View style={{ flexDirection: 'row' }}>
            <Text style={styles.pillsLoggedEditText}>Edit</Text>
            <FontAwesome
              name={'pencil'}
              style={styles.pillsLoggedEditTextIcon}
            />
          </View>
        </View>
      </View>
      {infoArray.map(entry => (
        <View
          key={entry.id}
          style={{
            flexDirection: 'row',
            marginLeft: 3,
            marginRight: 3,
          }}
        >
          <View
            style={{
              flexDirection: 'column',
              alignItems: 'flex-start',
              width: '45%',
            }}
          >
            <Text style={styles.pillsLoggedInfoText}>{entry.name}</Text>
          </View>

          <View
            style={{
              flexDirection: 'column',
              alignItems: 'flex-start',
              width: '15%',
            }}
          >
            <Text style={styles.pillsLoggedInfoText}>{entry.dosage}</Text>
          </View>

          <View
            style={{
              flexDirection: 'column',
              alignItems: 'flex-end',
              width: '40%',
            }}
          >
            <Text style={styles.pillsLoggedInfoText}>
              {entry.formattedTimeTaken}
            </Text>
          </View>
        </View>
      ))}
    </View>
  );
}

PillsLoggedTodayCard.propTypes = {
  /**
   * The header that is displayed at the top to describe the purpose of the card.
   */
  title: PropTypes.string,
  /**
   * An array containing the information for the card.
   */
  infoArray: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string,
      dosage: PropTypes.string,
      formattedTimeTaken: PropTypes.string,
    })
  ),
};

const styles = StyleSheet.create({
  pillsLoggedInfoContainer: {
    marginTop: 10,
    marginBottom: 10,
    marginLeft: 10,
    marginRight: 10,
    height: 115,
    ...Platform.select({
      ios: {
        shadowColor: 'black',
        shadowOffset: { width: 0, height: -3 },
        shadowOpacity: 0.1,
        shadowRadius: 3,
      },
      android: {
        elevation: 5,
      },
    }),
    backgroundColor: '#fbfbfb',
    borderRadius: 10,
    paddingVertical: 15,
    paddingHorizontal: 15,
  },
  pillsLoggedTitleText: {
    marginBottom: 10,
    fontSize: 15,
    color: 'rgba(70,70,70, 1)',
    textAlign: 'center',
  },
  pillsLoggedInfoText: {
    fontSize: 13,
    color: 'rgba(90,90,90, 1)',
    textAlign: 'left',
  },
  pillsLoggedEditText: {
    fontSize: 14,
    fontWeight: 'bold',
    color: 'rgba(121,51,153, 1)',
    textAlign: 'left',
  },
  pillsLoggedEditTextIcon: {
    fontSize: 14,
    fontWeight: 'bold',
    color: 'rgba(121,51,153, 1)',
    textAlign: 'left',
    marginTop: 2,
    marginLeft: 5,
  },
});
