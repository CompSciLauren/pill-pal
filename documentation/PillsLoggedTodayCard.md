## PillsLoggedTodayCard

prop | type | default | required | description
---- | :----: | :-------: | :--------: | -----------
**infoArray** | `Array[]<Shape>` |  | :x: | An array containing the information for the card.
**infoArray[].dosage** | `String` |  | :x: | 
**infoArray[].formattedTimeTaken** | `String` |  | :x: | 
**infoArray[].name** | `String` |  | :x: | 
**title** | `String` |  | :x: | The header that is displayed at the top to describe the purpose of the card.

