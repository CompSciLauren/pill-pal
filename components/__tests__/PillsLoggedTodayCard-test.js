import React from 'react';
import renderer from 'react-test-renderer';

import { PillsLoggedTodayCard } from '../PillsLoggedTodayCard';

// mock data to represent pills the user has logged so far today
const pillsLoggedToday = [
  {
    id: 0,
    name: 'Ibuprofen',
    dosage: '2 pills',
    formattedTimeTaken: '8:30 a.m.',
  },
  {
    id: 1,
    name: 'Ibuprofen',
    dosage: '2 pills',
    formattedTimeTaken: '1:07 p.m.',
  },
  {
    id: 2,
    name: 'Nitroglycerin',
    dosage: '1 pill',
    formattedTimeTaken: '3:45 p.m.',
  },
];

it(`renders correctly`, () => {
  const tree = renderer
    .create(
      <PillsLoggedTodayCard
        title="Pills Logged Today"
        infoArray={pillsLoggedToday}
      />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
