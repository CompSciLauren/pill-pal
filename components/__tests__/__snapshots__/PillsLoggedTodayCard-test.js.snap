// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fbfbfb",
      "borderRadius": 10,
      "height": 115,
      "marginBottom": 10,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 10,
      "paddingHorizontal": 15,
      "paddingVertical": 15,
      "shadowColor": "black",
      "shadowOffset": Object {
        "height": -3,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 3,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "marginLeft": 30,
          "width": "78%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(70,70,70, 1)",
            "fontSize": 15,
            "marginBottom": 10,
            "textAlign": "center",
          }
        }
      >
        Pills Logged Today
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexDirection": "column",
          "width": "22%",
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "rgba(121,51,153, 1)",
              "fontSize": 14,
              "fontWeight": "bold",
              "textAlign": "left",
            }
          }
        >
          Edit
        </Text>
        <Text />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginLeft": 3,
        "marginRight": 3,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexDirection": "column",
          "width": "45%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(90,90,90, 1)",
            "fontSize": 13,
            "textAlign": "left",
          }
        }
      >
        Ibuprofen
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexDirection": "column",
          "width": "15%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(90,90,90, 1)",
            "fontSize": 13,
            "textAlign": "left",
          }
        }
      >
        2 pills
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flexDirection": "column",
          "width": "40%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(90,90,90, 1)",
            "fontSize": 13,
            "textAlign": "left",
          }
        }
      >
        8:30 a.m.
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginLeft": 3,
        "marginRight": 3,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexDirection": "column",
          "width": "45%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(90,90,90, 1)",
            "fontSize": 13,
            "textAlign": "left",
          }
        }
      >
        Ibuprofen
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexDirection": "column",
          "width": "15%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(90,90,90, 1)",
            "fontSize": 13,
            "textAlign": "left",
          }
        }
      >
        2 pills
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flexDirection": "column",
          "width": "40%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(90,90,90, 1)",
            "fontSize": 13,
            "textAlign": "left",
          }
        }
      >
        1:07 p.m.
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginLeft": 3,
        "marginRight": 3,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexDirection": "column",
          "width": "45%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(90,90,90, 1)",
            "fontSize": 13,
            "textAlign": "left",
          }
        }
      >
        Nitroglycerin
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexDirection": "column",
          "width": "15%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(90,90,90, 1)",
            "fontSize": 13,
            "textAlign": "left",
          }
        }
      >
        1 pill
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flexDirection": "column",
          "width": "40%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(90,90,90, 1)",
            "fontSize": 13,
            "textAlign": "left",
          }
        }
      >
        3:45 p.m.
      </Text>
    </View>
  </View>
</View>
`;
